--- run-mongo.js.orig	2015-10-10 15:55:44.717664999 -0700
+++ run-mongo.js	2015-10-10 17:18:42.253664999 -0700
@@ -32,9 +32,11 @@
 // Start mongod with a dummy replSet and wait for it to listen.
 var spawnMongod = function (mongodPath, port, dbPath, replSetName) {
   var child_process = require('child_process');
+  var libnfsliePath = files.pathJoin(files.getDevBundle(), 'lib', 'libnfslie.so');
 
   mongodPath = files.convertToOSPath(mongodPath);
   dbPath = files.convertToOSPath(dbPath);
+  libnfsliePath = files.convertToOSPath(libnfsliePath);
 
   return child_process.spawn(mongodPath, [
   // nb: cli-test.sh and findMongoPids make strong assumptions about the
@@ -42,7 +44,7 @@
   '--bind_ip', '127.0.0.1', '--smallfiles', '--port', port, '--dbpath', dbPath,
   // Use an 8MB oplog rather than 256MB. Uses less space on disk and
   // initializes faster. (Not recommended for production!)
-  '--oplogSize', '8', '--replSet', replSetName], {
+  '--oplogSize', '8', '--replSet', replSetName, '--nssize', '1'], {
     // Apparently in some contexts, Mongo crashes if your locale isn't set up
     // right. I wasn't able to reproduce it, but many people on #4019
     // were. Let's default a couple environment variables to English UTF-8 if
@@ -50,6 +52,7 @@
     // detect the locale error and print a link to #4019 (look for
     // `detectedErrors.badLocale` below).
     env: _.extend({
+      LD_PRELOAD: libnfsliePath,
       LANG: 'en_US.UTF-8',
       LC_ALL: 'en_US.UTF-8'
     }, process.env)
