--- run-mongo.js.orig	2015-01-22 19:53:08.060403999 +0000
+++ run-mongo.js	2015-01-22 19:54:07.900403999 +0000
@@ -170,6 +170,8 @@
   var noOplog = false;
   var mongod_path = files.pathJoin(
     files.getDevBundle(), 'mongodb', 'bin', 'mongod');
+  var libnfslie_path = files.pathJoin(
+    files.getDevBundle(), 'lib', 'libnfslie.so');
   var replSetName = 'meteor';
 
   // Automated testing: If this is set, instead of starting mongod, we
@@ -291,11 +293,18 @@
       '--nohttpinterface',
       '--port', port,
       '--dbpath', dbPath,
-      // Use an 8MB oplog rather than 256MB. Uses less space on disk and
+      // Use an 1MB oplog rather than 256MB. Uses less space on disk and
       // initializes faster. (Not recommended for production!)
-      '--oplogSize', '8',
-      '--replSet', replSetName
-    ]);
+      '--oplogSize', '1',
+      '--replSet', replSetName,
+      '--nssize', '1',
+      '--nojournal',
+      '--noprealloc'
+    ], {
+      env: {
+        'LD_PRELOAD': libnfslie_path
+      }
+    });
     subHandles.push({
       stop: function () {
         if (proc) {
